corehq/apps/ivr/migrations/0002_call_app_id.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.22 on 2019-10-28 20:52
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+from corehq.apps.ivr.models import Call
+from corehq.apps.app_manager.dbaccessors import get_apps_in_domain
+from corehq.util.django_migrations import skip_on_fresh_install
+
+
+app_id_by_form_unique_id = {}
+
+
+def _update_model(model, domain):
+    if model.form_unique_id is None:
+        return None
+
+    if model.form_unique_id not in app_id_by_form_unique_id:
```

Comment:
I think this might make more sense to populate `app_id_by_form_unique_id` once per domain (could make a class and make this a memoized getter with `domain` as the sole param).

As-is, this could do the expensive operation many times per domain if there are `form.unique_id` values that weren't found.

Reply:
Good point. Updated in a818a1528320a5b5acd74f7bc55014599eca9c76 (at this point I admit I'm a little tired of updating all four of these migrations, but not tired enough to DRY up the code...would have done it by now if they weren't migrations in different apps).